### PR TITLE
adds installation instructions for pypatchmatch on Linux

### PR DESCRIPTION
**Pull request by lstein**
This PR adds installation instructions for pypatchmatch on Ubuntu Linux. We need instructions for Macintosh and non-debian Linux as well.

### CODE REVIEW

File: docs/installation/INSTALL_PATCHMATCH.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+Unfortunately, it is a C++ optimized module and installation
+can be somewhat challenging. This guide leads you through the steps.
+
+## Windows
+
+You're in luck! On Windows platforms PyPatchMatch will install
+automatically on Windows systems with no extra intervention.
+
+## Linux
```
Comment (Kyle0654) [inline docs/installation/INSTALL_PATCHMATCH.md:19]: Might want to note that these instructions are for Ubuntu (and I assume Debian?), where apt is available.

**Review by Kyle0654 — APPROVED** [review]

File: docs/installation/INSTALL_PATCHMATCH.md (hunk 1 above)
Comment (lstein) [reply docs/installation/INSTALL_PATCHMATCH.md:19]: I have clarified that the recipe is for Debian, Ubuntu and Mint.

Now the PR needs another approval!

**Comment by psychedelicious** [comment]
I'm confused here...

I have compiled `libpatchmatch.so` against macOS arm-64 and it appears that this is sufficient to use patchmatch. @mauwii copied it to their system and it worked. Did we decide not to just ship the compiled libraries? The actual installation instruction (to set up the environment and compile it) were pretty tedious.

**Comment by lstein** [comment]
> I'm confused here...
> 
> I have compiled `libpatchmatch.so` against macOS arm-64 and it appears that this is sufficient to use patchmatch. @mauwii copied it to their system and it worked. Did we decide not to just ship the compiled libraries? The actual installation instruction (to set up the environment and compile it) were pretty tedious.

Perfect! Let's ship it and then we won't need the installation instructions. Can you confirm that libpatchmatch.so has no dependencies on external libraries that need to be installed on the Mac? On my Ubuntu system, the same file needs various opencv .so files.

Do you know how to add libpatchmatch.so to the pypatchmatch package so that it is correctly identified and downloaded?

**Comment by mauwii** [comment]
> FWIW, I added `patchmatch` to the Docker image in https://github.com/invoke-ai/InvokeAI/pull/1544/files using the Linux installation instructions, and it seems to have worked as expected. But that does add ~900MB of largely unnecessary `opencv` dependencies, so having a compiled library would be beneficial.

Wouldn't it work to create the library in a separate stage and just copy over the compiled version of it?

**Comment by ebr** [comment]
> Wouldn't it work to create the library in a separate stage and just copy over the compiled version of it?

See my response in the other PR - I tried a few ways to attempt that approach, but couldn't quite nail it down. I'll come back to optimize it another time, unless we build and distribute the compiled library as discussed above.

**Comment by lstein** [comment]
superseded

**Pull request closed without merging**